src/Entity/EntityDocument.php

```diff
@@ -31,4 +33,15 @@ public function getId();
 	 */
 	public function getType();
 
+	/**
+	 * Sets the id of the entity. If the id is not of the correct type,
+	 * an exception will be thrown. A specific derivative of EntityId is
+	 * always supported.
+	 *
+	 * @param EntityId|mixed $id
```

Comment:
I suggest something like `mixed|EntityId|int|null`, so it can be `ItemId|int|null` in the subclasses without violating LSP.